view/src/main/java/dev/marcellogalhardo/retained/view/ViewRetainedObject.kt

```diff
@@ -0,0 +1,52 @@
+package dev.marcellogalhardo.retained.view
+
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import androidx.core.os.bundleOf
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.findViewTreeLifecycleOwner
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
+import androidx.navigation.NavBackStackEntry
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.findViewTreeSavedStateRegistryOwner
+import dev.marcellogalhardo.retained.core.InternalRetainedApi
+import dev.marcellogalhardo.retained.core.RetainedEntry
+import dev.marcellogalhardo.retained.core.createRetainedObjectLazy
+
+/**
+ * Returns a [Lazy] delegate to access a retained object by **default** scoped to this
+ * [View] (e.g., [NavBackStackEntry], [Fragment] or [Activity]).
+ *
+ * ```
+ * class MyView(context: Context) : View(context) {
+ *     val vm by retain { ViewModel() }
+ * }
+ * class ViewModel(val name: String = "")
+ * ```
+ *
+ * This property can be accessed only after the [LifecycleOwner] is ready to use Jetpack
+ * [ViewModel], and access prior to that will result in [IllegalArgumentException]. TODO: right now it actually throws a NPE
+ *
+ * @see createRetainedObject
+ */
+@OptIn(InternalRetainedApi::class)
+inline fun <reified T : Any> View.retain(
+    key: String = T::class.java.name,
```

Comment:
How would we handle multiple Views in the same LifecycleOwner? This implementation would make all views share the same retained instance. One idea is to generate a unique identifier when creating the retained object lazy:

```kotlin
reateRetainedObjectLazy("$id-key", getViewModelStoreOwner, getSavedStateRegistryOwner, getDefaultArgs, createRetainedObject)
```

However, if we decide to do something like that, we would need to be sure the view has time to restore the ID in a configuration change or process death to ensure we will access the right view model.

Reply:
Very good point. Is this an also valid concern for the fragment case when we have multiple fragments in the same activity?

But yeah, we'd have to test if using the `id` there would work. I'm thinking if assigning and using a random `tag` would be any better, but I honestly don't know and would have to test it as well.

Reply:
Each `Fragment` is a `LifecycleOwner` itself, and AndroidX will already handle it for us doing the required scope. Attaching two fragments with the same retained instance would create different instances per fragment.

For the random tag, you would need to preserve it on configuration changes to restore the correct retained instance. As the view system already provides a unique identifier, I guess it is safer to just use it. 🤔

Reply:
Of course, I forget views are not the owner themselves, thanks!

And about the tag, I assumed they were automatically retained, that's why I brought them up, but I might be wrong. But I agree using the `id` is a better option if it works fine, and I'd be willing to explore tags only if that's not the case.

Reply:
Change default key value from class name to id https://github.com/marcellogalhardo/retained/pull/6/commits/e8e9fdec107472f0e00e94c9e5e56b4a596afaab